evm/src/generation/mod.rs

```diff
@@ -45,18 +46,28 @@ pub fn generate_traces<F: RichField + Extendable<D>, const D: usize>(
         current_cpu_row,
         memory,
         keccak_inputs,
-        logic_ops: logic_inputs,
+        logic_ops,
+        prover_inputs,
         ..
     } = state;
     assert_eq!(current_cpu_row, [F::ZERO; NUM_CPU_COLUMNS].into());
+    assert_eq!(prover_inputs, vec![], "Not all prover inputs were consumed");
 
     let cpu_trace = trace_rows_to_poly_values(cpu_rows);
     let keccak_trace = all_stark.keccak_stark.generate_trace(keccak_inputs);
-    let logic_trace = all_stark.logic_stark.generate_trace(logic_inputs);
+    let logic_trace = all_stark.logic_stark.generate_trace(logic_ops);
     let memory_trace = all_stark.memory_stark.generate_trace(memory.log);
     vec![cpu_trace, keccak_trace, logic_trace, memory_trace]
 }
 
-fn generate_txn<F: Field>(_state: &mut GenerationState<F>, _txn: &TransactionData) {
-    todo!()
+fn generate_txn<F: Field>(state: &mut GenerationState<F>, txn: &TransactionData) {
+    // TODO: Add transaction RLP to prover_input.
+
+    // Supply Merkle trie proofs as prover inputs.
+    for proof in &txn.trie_proofs {
+        let proof = proof
+            .iter()
+            .flat_map(|node_rlp| node_rlp.iter().map(|byte| U256::from(*byte)));
```

Comment:
Hm it would mean a lot less memory operations, but we'd need to do shifts and masks in the RLP parser to extract bytes (or extract 256-bit and 160-bit fields which won't generally be aligned). I'll look into the costs a bit.